src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.spec.ts

```diff
@@ -104,7 +104,35 @@ describe('RbdTrashListComponent', () => {
   });
 
   describe('display purge button', () => {
-    beforeEach(() => {});
+    let images1: any[];
+    const addImage = (id: string) => {
+      images1.push({
+        id: id,
+        pool_name: 'pl',
+        deferment_end_time: 'abc'
+      });
+    };
+
+    beforeEach(() => {  
+      summaryService['summaryDataSource'].next({ executingTasks: [] });
+      spyOn(rbdService, 'listTrash').and.callFake(() => {
+        of( [{ poool_name: 'rbd', status: 1, value: images1  }])
+      });
+      fixture.detectChanges();
+    });
+    
+    it('should show button disabled when no image is in trash', () => {
+      expect(component.disablePurgeBtn).toBeFalsy();
+    });
+
+    it('should show button enabled when an existing image is in trash', () => {
+      images1 = [];
+      addImage('1');
+      let var1 = [{poool_name: 'rbd', status: 1, value: images1}];
```

Comment:
Typo in "poool_name".
Instead of "var1", it would be preferable a more meaningful name, like "payload" or sth similar.

Reply:
Done

Reply:
Made all recommended changes